### PR TITLE
liquid tx support

### PR DESCRIPTION
**Pull request by glslang**

### CODE REVIEW

File: src/transaction.c [hunk 1]
```diff
@@ -358,36 +725,375 @@ int wally_tx_input_free(struct wally_tx_input *input)
...
+#endif /* BUILD_ELEMENTS */
+    if ((src->script_len && !new_script)
+#ifdef BUILD_ELEMENTS
+        || (src->asset_len && !new_asset) ||
+           (src->value_len && !new_value) ||
+           (src->nonce_len && !new_nonce) ||
+           (src->surjectionproof_len && !new_surjectionproof) ||
+           (src->rangeproof_len && !new_rangeproof)
+#endif /* BUILD_ELEMENTS */
+    ) {
```
Comment (jkauffman1) [inline src/transaction.c]: Missing wally_free(new_script)

File: src/transaction.c [hunk 2]
```diff
@@ -358,36 +725,375 @@ int wally_tx_input_free(struct wally_tx_input *input)
...
+    size_t surjectionproof_len,
+    const unsigned char *rangeproof,
+    size_t rangeproof_len)
+{
+
+    if (((surjectionproof != NULL) != (surjectionproof_len != 0)) ||
+       ((rangeproof != NULL) != (rangeproof_len != 0)))
+        return WALLY_EINVAL;
+#ifdef BUILD_ELEMENTS
+    if (output->surjectionproof)
```
Comment (jkauffman1) [inline src/transaction.c]: nit: This check of the pointer before calling wally_free seems inconsistent with other places in the file

File: src/transaction.c [hunk 3]
```diff
@@ -358,36 +725,375 @@ int wally_tx_input_free(struct wally_tx_input *input)
...
+                                        asset, asset_len,
+                                        value, value_len,
+                                        nonce, nonce_len,
+                                        surjectionproof,
+                                        surjectionproof_len,
+                                        rangeproof, rangeproof_len,
+                                        result);
+    if (ret != WALLY_OK) {
+        clear_and_free(result, sizeof(*result));
+        *output = NULL;
```
Comment (jkauffman1) [inline src/transaction.c]: Free output?

File: src/transaction.c [hunk 4]
```diff
@@ -552,10 +1348,11 @@ int wally_tx_remove_input(struct wally_tx *tx, size_t index)
...
 {
     uint64_t total;
-
-    if (!is_valid_tx(tx) || !is_valid_tx_output(output) ||
-        wally_tx_get_total_output_satoshi(tx, &total) != WALLY_OK ||
-        total + output->satoshi < total || total + output->satoshi > WALLY_SATOSHI_MAX)
+    const bool is_elements = output->features & WALLY_TX_IS_ELEMENTS;
+    if (!is_valid_tx(tx) || (!is_valid_tx_output(output) && !is_valid_elements_tx_output(output)) ||
+        ((wally_tx_get_total_output_satoshi(tx, &total) != WALLY_OK ||
+          (total + output->satoshi < total || total + output->satoshi > WALLY_SATOSHI_MAX)) &&
```
Comment (jkauffman1) [inline src/transaction.c]: Was it intended to exclude elements from the overflow check as well as the > WALLY_SATOSHI_MAX check?

**Comment by jgriffiths** [comment]
Merged with transaction.c uncrustified, thanks!

**Pull request closed without merging**